src/runtime/buffer_t.cpp

```diff
@@ -114,6 +114,22 @@ halide_buffer_t *_halide_buffer_init_from_buffer(halide_buffer_t *dst,
     return dst;
 }
 
+HALIDE_BUFFER_HELPER_ATTRS
+halide_buffer_t *_halide_buffer_crop(halide_buffer_t *dst,
+                                     halide_dimension_t *dst_shape,
+                                     const halide_buffer_t *src,
+                                     const int *min, const int *extent) {
+    *dst = *src;
+    dst->dim = dst_shape;
+    for (int i = 0; i < dst->dimensions; i++) {
+        dst->dim[i] = src->dim[i];
+        dst->dim[i].min = min[i];
+        dst->dim[i].extent = extent[i];
+        dst->host += (min[i] - src->dim[i].min) * src->dim[i].stride * src->type.bytes();
```

Comment:
This does a bunch of extra multiplies. Consider:

   int64_t offset = 0;
   for (int i ...) {
      offset += (min[i] - src->dim[i].min) * src->dim[i].stride;
   }
   dst->host += offset * src->type.bytes();